### DISCUSSION TITLE
Can't manage or authenticate with LDAP users

### DISCUSSION THREAD

**Discussion by hectorgonzalezdeandres** [DependencyTrack/dependency-track#983] [General]
Hi, I am installing a 4.1 version of dependency track, and basically my problem is similar to this one:

https://github.com/DependencyTrack/dependency-track/issues/154

I can't see errors in logs, but I don´t know how to set debug mode in pod of k8s (https://github.com/DependencyTrack/dependency-track/discussions/973)

Basically I have done my config of ldap and all is ok in INFO level log mode. But when I create a LDAP user in my deptrack's console the state appears invalid. Which is the correct format to create this user?

If I test connection to LDAP with Ldapsearch works fine, so I think that problem could be in the use creation format. Could you provide me an example? I cant see it in documentation....

Regards!

**Comment by stevespringett** [comment]
Examples that are known to work:
https://docs.dependencytrack.org/getting-started/ldap-configuration/

**Reply by hectorgonzalezdeandres** [reply]
But this is an example of apiserver configuration

What I asked is a config of ui of dependency track. My apiserver conf is ok, like i said i've tested with ldapsearch.

Regards

**Reply by stevespringett** [reply]
Precisely. You asked:
> Which is the correct format to create this user?

The format to create the user is defined in the LDAP configuration.
```
alpine.ldap.auth.username.format=%s@example.com
```
This states that the format to create the user would simple be their username `john.doe`. The `@example.com` will be appended automatically.

**Reply by hectorgonzalezdeandres** [reply]
Thank you, but just it's the problem, I have tested creating john.doe alone john.doe@example.com and all posibilities. 

And state is the same syncing and later invalid, and no errors in logs.

Any help appreciate.

Regards!

**Comment by stevespringett** [comment]
LDAP users are created with an invalid state, but usually only takes a few seconds to the server to sync. When syncing DT will perform an LDAP search to find the users DN. If its found, then `invalid` will be replaced with the users DN. If its not found, then `invalid` will remain.

This tells me the LDAP search filter is likely incorrect or some other issue is happening. Enable debug to help diagnose the issue. I cannot help with K8s, but for Docker, its a simple matter of setting the environment variable `LOGGING_LEVEL` to `DEBUG` .

**Reply by hectorgonzalezdeandres** [reply]
Thanks, I'll try it with docker. 

Isn't any support for k8s?

Regards!

**Reply by stevespringett** [reply]
K8s isn't directly supported by the project, but there are community provided Helm charts available and many organizations run in that environment.

**Reply by hectorgonzalezdeandres** [reply]
I have been able to put logs in debug mode, but state is Syncing...

And no error in logs.

There is no relevant info in logs? What should I search?

Thanks!

--

**Reply by hectorgonzalezdeandres** [reply]
Hi, is there any support to help me to configure LDAP?

Regards

**Reply by stevespringett** [reply]
Search for messages from LdapSyncTask, especially messages with `Error occurred during LDAP synchronization`

Have you tried the same LDAP search filters using something like Apache Directory Studio?
